homeassistant/components/config/label_registry.py

```diff
@@ -0,0 +1,129 @@
+"""Websocket API to interact with the label registry."""
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.components.websocket_api.connection import ActiveConnection
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.label_registry import LabelEntry, async_get
+
+
+@callback
+def async_setup(hass: HomeAssistant) -> bool:
+    """Register the Label Registry WS commands."""
+    websocket_api.async_register_command(hass, websocket_list_labels)
+    websocket_api.async_register_command(hass, websocket_create_label)
+    websocket_api.async_register_command(hass, websocket_delete_label)
+    websocket_api.async_register_command(hass, websocket_update_label)
+    return True
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "config/label_registry/list",
+    }
+)
+@callback
+def websocket_list_labels(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle list labels command."""
+    registry = async_get(hass)
+    connection.send_result(
+        msg["id"],
+        [_entry_dict(entry) for entry in registry.async_list_labels()],
+    )
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "config/label_registry/create",
+        vol.Required("name"): str,
+        vol.Optional("color"): vol.Any(str, None),
```

Comment:
Should we enforce hex format?

Reply:
I guess we could require color as well?

Reply:
Done in [9c7748b](https://github.com/home-assistant/core/pull/110821/commits/9c7748bf88732f594a2273cafd433c64b6466a75)